### ISSUE TITLE
Rule Enhancement Proposal: "object-curly-newline" add configuration for named imports and exports

### ISSUE THREAD

**Issue by ikokostya** [eslint/eslint#7319]
ESLint version `v3.5.0`

Rule [object-curly-newline](http://eslint.org/docs/rules/object-curly-newline) currently supports:
- `"ObjectExpression"` configuration for object literals;
- `"ObjectPattern"` configuration for object patterns of destructuring assignments.
## Proposal

I propose to add similar configuration for named imports and exports:

``` js
{
    "object-curly-newline": ["error", {
        "ImportDeclaration": { "multiline": true },
        "ExportDeclaration": { "multiline": true }
    }]
}
```

Possible values:
### always

Examples of **incorrect** code for this rule with the `"always"` option:

``` js
import {foo, bar} from 'foo-bar';
import {foo as f, bar} from 'foo-bar';
import {foo,
    bar} from 'foo-bar';

export {foo, bar};
export {foo as f, bar};
export {foo,
   bar};
```

Examples of **correct** code for this rule with the `"always"` option:

``` js
import {
    foo,
    bar
} from 'foo-bar';
import {
    foo as f,
    bar
} from 'foo-bar';

export {
  foo,
  bar
};
export {
  foo as f,
  bar
};
```
### never

Examples of **incorrect** code for this rule with the `"never"` option:

``` js
import {
    foo,
    bar
} from 'foo-bar';

export {
   foo,
   bar
};
```

Examples of **correct** code for this rule with the `"never"` option:

``` js
import {foo, bar} from 'foo-bar';

export {foo, bar};
```
### multiline

Examples of **incorrect** code for this rule with the default `{ "multiline": true }` option:

``` js
import {
    foo, bar} from 'foo-bar';
import {foo, bar
    } from 'foo-bar';
import {
    foo, bar
} from 'foo-bar';

export {
   foo, bar};
export {
   foo, bar
};
```

Examples of **correct** code for this rule with the default `{ "multiline": true }` option:

``` js
import {foo, bar} from 'foo-bar';
import {foo as f, bar} from 'foo-bar';
import {
    foo,
    bar
} from 'foo-bar';
import {
    foo as f,
    bar
} from 'foo-bar';

export {
   foo,
   bar
};
```
### minProperties

Similar to current [`minProperties` behavior](http://eslint.org/docs/rules/object-curly-newline#minproperties). Maybe this option should be renamed to `minImportSpecifiers` or similar.
## Question

Maybe this should be separate rule instead of configuration of `object-curly-newline` rule.

**Comment by kaicataldo** [comment]
I'm for the ability to enforce these styles, but I do think it's a little strange to have named `import` and `export` statements covered by this rule since they're not really objects. That being said, we do have precedence for this, as we currently check named `import` and `export` statements in [object-curly-spacing](http://eslint.org/docs/rules/object-curly-spacing).

**Comment by evilebottnawi** [comment]
@kaicataldo what status? What can I do PR? And what names do better?

**Comment by kaicataldo** [comment]
@evilebottnawi Hey, thanks for being willing to make a PR. This issue hasn't been accepted yet, which means that the team hasn't decided whether or not this should be included in ESLint core. If and when this is accepted, we'd love to have your contribution!

@eslint/eslint-team Looks like @evilebottnawi might be willing to implement this should it be accepted. Can we get some opinions/votes for its inclusion?

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.

**Comment by platinumazure** [comment]
I don't remember seeing this issue, so I've just given it a :+1:. If we can get a champion from the ESLint team who didn't :+1: this, or if we can get one more :+1: and a champion from someone who did :+1: this, then we could reopen and accept the issue.

@eslint/eslint-team Is anyone willing to champion this issue, or should we leave it closed? Seems @evilebottnawi is willing to make a PR if we accept this.
